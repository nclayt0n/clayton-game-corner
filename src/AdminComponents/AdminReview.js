import React from 'react';

class AdminReview extends React.Component{
    render(){
        return(
        <>
            <section>
                <form>
                    <label> Title: 
                        <input type="text" placeholder='Portal'/>
                    </label>
                    <label>Images:<br/>
                        <input type="text" placeholder='/images/portal.jpg' />
                    </label>
                    <label>Review:<br/>
                        <textarea placeholder= 'Portal was a super awesome game' >
                        </textarea>
                    </label>
                    <label>Link to Buy:
                        <input type='url' placeholder="https://www.amazon.com/Portal-PC/dp/B00140P9G0" target="_blank"/>
                    </label>
                    <select>
                        <option >Type</option>
                        <option value='video'>Video</option>
                        <option value='tabletop'>Tabletop</option>
                    </select>
                </form>
                <button>Update</button>
                <button>Delete</button>
            </section>
            <section>
                <form>
                    <label> Title: 
                        <input type="text" placeholder='Splendor'/>
                    </label>
                    <label>Images:<br/>
                        <input type="text" placeholder='/images/BoardGameSplendor.jpg' />
                    </label>
                    <label>Review:<br/>
                        <textarea placeholder= 'Splendor was a super awesome game' ></textarea>
                    </label>
                    <label>Link to Buy:
                        <input type='url' placeholder='https://www.amazon.com/Asmodee-SPL01-Splendor/dp/B00IZEUFIA'/>
                    </label>
                    <select>
                        <option >Type</option>
                        <option value='video'>Video</option>
                        <option value='tabletop'>Tabletop</option>
                    </select>
                </form>
                <button>Update</button>
                <button>Delete</button>
            </section>
        </>)
    }
}
export default AdminReview;