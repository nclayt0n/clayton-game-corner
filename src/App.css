/* http://meyerweb.com/eric/tools/css/reset/ 
   v2.0 | 20110126
   License: none (public domain)
*/

html,
body,
div,
span,
applet,
object,
iframe,
h1,
h2,
h3,
h4,
h5,
h6,
p,
blockquote,
pre,
a,
abbr,
acronym,
address,
big,
cite,
code,
del,
dfn,
em,
img,
ins,
kbd,
q,
s,
samp,
small,
strike,
strong,
sub,
sup,
tt,
var,
b,
u,
i,
center,
dl,
dt,
dd,
ol,
ul,
li,
fieldset,
form,
label,
legend,
table,
caption,
tbody,
tfoot,
thead,
tr,
th,
td,
article,
aside,
canvas,
details,
embed,
figure,
figcaption,
footer,
header,
hgroup,
menu,
nav,
output,
ruby,
section,
summary,
time,
mark,
audio,
video {
    margin: 0;
    padding: 0;
    border: 0;
    font-size: 100%;
    font: inherit;
    vertical-align: baseline;
}


/* HTML5 display-role reset for older browsers */

article,
aside,
details,
figcaption,
figure,
footer,
header,
hgroup,
menu,
nav,
section {
    display: block;
}

body {
    line-height: 1;
}

ol,
ul {
    list-style: none;
}

blockquote,
q {
    quotes: none;
}

blockquote:before,
blockquote:after,
q:before,
q:after {
    content: '';
    content: none;
}

table {
    border-collapse: collapse;
    border-spacing: 0;
}

:root {
    --mainColor: rgb(17, 72, 82);
}

img {
    width: 200px;
    height: 200px;
}

main {
    margin-bottom: 150px;
}

nav {
    background-color: white;
}

article,
section {
    display: flex;
    flex-direction: column;
    text-align: center;
    border: 1px solid black;
    width: 300px;
    margin: 80px auto;
    padding: 15px;
}

p {
    padding: 10px;
    margin: auto;
}

textarea {
    border: 1px solid black;
}

footer {
    position: fixed;
    bottom: 0;
    display: flex;
    width: 100%;
    background-color: white;
}

h2,
h1 {
    font-size: 22px;
    text-align: center;
}

h3 {
    margin: 20px auto;
    text-align: center;
}

footer address {
    margin: auto;
}

form {
    display: flex;
    flex-direction: column;
    flex: 1;
}

form textarea {
    height: 150px;
}

button {
    border: 1px solid black;
    margin: auto;
    width: 150px;
    background-color: var(--mainColor);
    color: white;
}

header {
    position: fixed;
    top: 0;
    width: 100%;
    text-align: center;
    width: 100%;
    background-color: var(--mainColor);
}

aside {
    margin: 120px auto 20px;
}

#headerLogo {
    height: 90px;
    width: 80px;
    margin: 10px;
}

#loginContainer {
    margin-top: 150px;
}

#loginForm {
    width: 300px;
}

#loginForm input {
    width: 200px;
}

.notFoundPage {
    margin-top: 150px;
}