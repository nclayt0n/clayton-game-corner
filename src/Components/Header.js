import React from 'react';
import {withRouter} from 'react-router-dom';
import logo from '../images/smallCGCLogo.png';
class Header extends React.Component{
    render(){
        return(
        <>
            {(this.props.location.pathname==='/')
                ?''
                :(<header>
                    <img id='headerLogo'src={logo} alt='clayton game corner logo'/>
                </header>)
            }
        </>)
    }
}
export default withRouter(Header);