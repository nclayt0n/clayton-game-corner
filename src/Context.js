import React from 'react';

const Context = React.createContext({
    user_id: '',
    reviews: [],
    bio: '',
    upcomingGames: [],
    picture: '',
    addPicture: () => {},
    addReviews: () => {},
    addReview: () => {},
    deleteReview: () => {},
    updateReview: () => {},
    updateUpcomingGame: () => {},
    addBio: () => {},
    updateBio: () => {},
    addUpcomingGames: () => {},
    deleteUpcomingGame: () => {},
    addUpcomingGame: () => {},
});
export default Context;