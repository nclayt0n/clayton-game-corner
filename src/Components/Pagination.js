import React from 'react';
import Context from '../Context';

class Pagination extends React.Component{
    static contextType=Context;
    
    render(){
        return(
            <section className='pagination-button-container'>
                {this.props.page<1 
                    ? null
                    :<button 
                        aria-label='button for previous page'
                        className='button' 
                        type='button' 
                        onClick={()=>this.props.setPage(this.props.page-1)}
                        >Back
                    </button>}

                <button 
                        aria-label='current page'
                        className='button' 
                        id='currentPage' 
                        type='button'>Page {this.props.page+1}
                    </button>
                {this.props.items.length <this.props.pageLimit
                    ?null
                    :
                    <button 
                        aria-label='button for next page of users'
                        className='button' 
                        type='button' 
                        onClick={()=>this.props.setPage(this.props.page+1)}>Next
                    </button>
                    }
            </section>
        )
    }
}
export default Pagination;
