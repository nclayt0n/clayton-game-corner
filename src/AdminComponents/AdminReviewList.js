import React from 'react';
import AdminReview from './AdminReview';
import Pagination from '../Components/Pagination';
import {withRouter} from 'react-router-dom';
import { Receiver, UploadManager,UploadHandler } from 'react-file-uploader';
import ReactDOM from 'react-dom';
import DropzoneComponent from 'react-dropzone-component';
import '../../node_modules/dropzone/dist/min/dropzone.min.css';
import Nav from '../Components/Nav';
import Header from '../Components/Header';
import config from '../config';
import GameApiService from '../services/game-api-services';
import Context from '../Context';
import ValidationError from '../Validation/ValidationError';
let ReactDOMServer = require('react-dom/server');
const uuidv4=require('uuid');

class AdminReviewList extends React.Component{
    static contextType=Context;
    constructor(){
        super()
        this.state={
            error:'',
            page:0,
            pageLimit:10
        };
    }
    componentDidMount(){
        GameApiService.getApiCall(`${config.API_ENDPOINT}/game/review?limit=${this.state.pageLimit}&offset=${this.state.page*this.state.pageLimit}`)
        .then((reviews) => {
                    this.context.addReviews(reviews);
                })
                .catch(error => {
                    this.setState({ error });
                });
    }
    addReview=(e)=>{
        e.preventDefault();
        let title=e.target.title.value;
        let review=e.target.review.value;
        let game_type=e.target.game_type.value;
        let link=e.target.link.value;
        let picture='';
       
        if(review.length===0){
            this.setState({error:'Must include a game review'});
        }
        if(title.length===0){
            this.setState({error:'Must include a game title'});
        }
        if(game_type.length===0){
            this.setState({error:'Must choose a game type'});
        }if(review.length>0&&title.length>0&&game_type.length>0){
            this.setState({error:''});
            GameApiService.postReview(title,game_type,link,picture, review)
                .then((newReview) => {
                    this.context.addReview(newReview); this.props.history.push('/admin/game/review-list');
                })
                .catch(error => {
                    this.setState({ error:error.message });
                });
        }
    }
    setPage=(page)=>{
        this.setState({page:page})
        GameApiService.getApiCall(`${config.API_ENDPOINT}/game/review/tabletop?limit=${this.state.pageLimit}&offset=${page*this.state.pageLimit}`)
        .then((reviews) => {
                    this.context.addReviews(reviews);
                })
                .catch(error => {
                    this.setState({ error });
                });
    };
    render(){
      
        return(
        <>
        <Header/>
        <Nav/>
            <section key={uuidv4()}>
                <form onSubmit={(e)=>this.addReview(e,this.context)}>
                <fieldset>
                    <legend>Add Game Review
                    </legend>
                    <label htmlFor='title'>Title: 
                        <input name='title' type="text"/>
                    </label><br/>
                    <label htmlFor='review'>Review: <br/>
                        <textarea placeholder='add review here' name='review'></textarea>
                    </label><br/>
                    <label htmlFor='link'>Link to Buy: 
                        <input name='link' type="url"/>
                    </label><br/>
                    <select name='game_type'>
                        <option value=''>Game Type:</option>
                        <option value='video'>Video</option>
                        <option value='tabletop'>Tabletop</option>
                    </select>
                    <fieldset>
                        <legend> Images (max. 5):</legend>
                    </fieldset>
                </fieldset> 
                <button type='submit'>Add Review</button>
                </form>
                <ValidationError errorMessage={this.state.error}/>
            </section>
            

            {/* <Receiver
                customClass={'addReview'}
                isOpen={true}
                // onDragEnter={FUNCTION}
                // onDragOver={FUNCTION}
                // onDragLeave={FUNCTION}
                // onFileDrop={FUNCTION}
            >
                <div className='droppable'>
                    visual layer of the receiver (drag & drop panel)
                </div>
            </Receiver> */}

            <h2>Review List</h2>
            {this.context.reviews.map(review=>{
                    return <AdminReview key={review.id} review={review}/>
            })}
            {this.context.reviews.length===0
                ?<section>
                    <p>No Reviews to be displayed</p>
                </section>
                :<Pagination 
                    page={this.state.page} 
                    pageLimit={this.state.pageLimit} 
                    setPage={(page)=>this.setPage(page)} 
                    items={this.context.reviews}/>
            }
           
        </>)
    }
}
export default withRouter(AdminReviewList);