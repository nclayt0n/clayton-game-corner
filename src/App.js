import React from 'react';
import {Switch,Route} from 'react-router-dom';
import Context from './Context';
import LandingPage from './Components/LandingPage';
import PublicOnlyRoute from './Utils/PublicOnlyRoute';
import PrivateRoute from './Utils/PrivateRoute';
import TabletopGameReview from './Components/TabletopGameReviews';
import VideoGameReview from './Components/VideoGameReview';
import UpcomingGameList from './Components/UpcomingGameList';
import './App.css';
import NotFound from './Components/NotFound';
import Header from './Components/Header';
import Nav from './Components/Nav';
import Footer from './Components/Footer';
import UpcomingGameInfo from './Components/UpcomingGameInfo';
import AdminLandingPage from './AdminComponents/AdminLandingPage';
import AdminReview from './AdminComponents/AdminReviewList';
import AdminReviewList from './AdminComponents/AdminReviewList';
import AdminLogin from './AdminComponents/AdminLogin';
import AdminUpcomingGameList from './AdminComponents/AdminUpcomingGameList';
import TokenService from './services/token-service'

class App extends React.Component {
  constructor(){
    super()
      this.state={
          user_id:0,
          reviews:[],
          bio:'',
          upcomingGames:[]
      };
    }
    handleAddReviews=(reviews)=>{
    this.setState({
      reviews:[...reviews]
    })
  }
  handleAddUpcomings=(upcomingGames)=>{
    this.setState({
      upcomingGames:[...upcomingGames]
    })
  }
  handleAddBio=(bio)=>{
    this.setState({
      bio:bio
    })
  }
  render(){
    let userId;
    (TokenService.getAuthToken()===null)?userId=0
      :userId=TokenService.decodeAuthToken(TokenService.getAuthToken());
    
    const contextValue={
      user_id:userId,
      bio:this.state.bio,
      upcomingGames:this.state.upcomingGames,
      reviews:this.state.reviews,
      addReviews:this.handleAddReviews,
      addUpcomingGames:this.handleAddUpcomings,
      addBio:this.handleAddBio
    };
  return (
    <main className='App'>
        <Context.Provider value={contextValue}>
        <Switch>
          <Route exact path='/' component={LandingPage}/>
          <Route path='/game/review/tabletop' component={TabletopGameReview}/>
          <Route path='/game/review/video' component={VideoGameReview}/>
          <Route exact path='/game/upcoming' component={UpcomingGameList}/>
          <Route exact path='/game/upcoming/info' component={UpcomingGameInfo}/>
          <Route path='/nav' component={Nav}/>
          <PrivateRoute exact path='/admin' component={AdminLandingPage}/>
          <PrivateRoute exact path='/admin/game/review-list' component={AdminReviewList}/>
          <PrivateRoute exact path='/admin/game/upcoming-list' component={AdminUpcomingGameList}/>
          <Route exact path='/admin/login' component={AdminLogin}/>
          <Route component={NotFound}/>
        </Switch>
        </Context.Provider>
      <Footer/>
    </main>
  );
  }
}

export default App;