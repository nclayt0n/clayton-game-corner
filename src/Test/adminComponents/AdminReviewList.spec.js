import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter,StaticRouter} from 'react-router-dom';
import App from '../../App';
import Pagination from '../../Components/Pagination';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Context from '../../Context';
import AdminReviewList from '../../AdminComponents/AdminReviewList';
import Header from '../../Components/Header';
import AdminReview from '../../AdminComponents/AdminReview';

describe('Admin Review List ',()=>{
  it('renders without crashing', () => {
      let reviews = [{
            "title": "Test Review 1",
            "review": "This was the best game ever!",
            "picture": "['http://ecx.images-amazon.com/images/I/51l4YWeAEvL._SY300_.jpg']",
            "link": "http://www.notArealSite.com",
            "game_type": "tabletop"
        },
        {
            "title": "Test Review 2",
            "review": "This was an ok game.",
            "picture": "['http://ecx.images-amazon.com/images/I/51l4YWeAEvL._SY300_.jpg']",
            "link": "http://www.notArealSite.com",
            "game_type": "tabletop"
        }];
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <App>
                        <AdminReviewList key={6}>
                            <Header/>
                            <section key={626}>
                                <form onSubmit={(e)=>this.addReview(e,this.context)}>
                                <fieldset>
                                    <legend>Add Game Review
                                    </legend>
                                    <label htmlFor='title'>Title: 
                                        <input name='title' type="text"/>
                                    </label><br/>
                                    <label htmlFor='review'>Review: <br/>
                                        <textarea placeholder='add review here' name='review'></textarea>
                                    </label><br/>
                                    <label htmlFor='link'>Link to Buy: 
                                        <input name='link' type="url"/>
                                    </label><br/>
                                    <select name='game_type'>
                                        <option value=''>Game Type:</option>
                                        <option value='video'>Video</option>
                                        <option value='tabletop'>Tabletop</option>
                                    </select>
                                </fieldset> 
                                <button type='submit'>Add Review</button>
                                </form>
                            </section>
            
                            <h2>Review List</h2>
                            {reviews.map(review=>{
                                return <AdminReview key={78} review={review}/>
                            })}
                            <Pagination 
                                page={0} 
                                pageLimit={10} 
                                setPage={(page)=>this.setPage(page)} 
                                items={reviews}/>
                        </AdminReviewList>
                    </App>
                </BrowserRouter>,
                div
              );
            ReactDOM.unmountComponentAtNode(div);
        });
    it('snapshot', () => {
        let reviews = [{
            "title": "Test Review 1",
            "review": "This was the best game ever!",
            "picture": "['http://ecx.images-amazon.com/images/I/51l4YWeAEvL._SY300_.jpg']",
            "link": "http://www.notArealSite.com",
            "game_type": "tabletop"
        },
        {
            "title": "Test Review 2",
            "review": "This was an ok game.",
            "picture": "['http://ecx.images-amazon.com/images/I/51l4YWeAEvL._SY300_.jpg']",
            "link": "http://www.notArealSite.com",
            "game_type": "tabletop"
        }];
        const wrapper=shallow(
            <StaticRouter>
                <App>
                    <Context.Provider>
                        <AdminReviewList key={6}>
                            <Header/>
                            <section key={301}>
                                <form onSubmit={(e)=>this.addReview(e,this.context)}>
                                    <fieldset>
                                        <legend>Add Game Review</legend>
                                        <label htmlFor='title'>Title: 
                                            <input name='title' type="text"/>
                                        </label><br/>
                                        <label htmlFor='review'>Review: <br/>
                                            <textarea placeholder='add review here' name='review'></textarea>
                                        </label><br/>
                                        <label htmlFor='link'>Link to Buy: 
                                            <input name='link' type="url"/>
                                        </label><br/>
                                        <select name='game_type'>
                                            <option value=''>Game Type:</option>
                                            <option value='video'>Video</option>
                                            <option value='tabletop'>Tabletop</option>
                                        </select>
                                    </fieldset> 
                                    <button type='submit'>Add Review</button>
                                    </form>
                                </section>
            
                                <h2>Review List</h2>
                                {reviews.map(review=>{
                                    return <AdminReview key={288} review={review}/>
                                })}
                                <Pagination 
                                    page={0} 
                                    pageLimit={10} 
                                    setPage={(page)=>this.setPage(page)} 
                                    items={reviews}/>
                        </AdminReviewList>
                    </Context.Provider>
                </App>
            </StaticRouter>)
                        expect(toJson(wrapper))
                        .toMatchSnapshot()
    });
})