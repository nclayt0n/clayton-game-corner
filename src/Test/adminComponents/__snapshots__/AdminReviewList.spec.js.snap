// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Review List  snapshot 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
  staticContext={Object {}}
>
  <withRouter(App)>
    <ContextProvider>
      <withRouter(AdminReviewList)
        key="6"
      >
        <Header />
        <section
          key="301"
        >
          <form
            onSubmit={[Function]}
          >
            <fieldset>
              <legend>
                Add Game Review
              </legend>
              <label
                htmlFor="title"
              >
                Title:
                <input
                  name="title"
                  type="text"
                />
              </label>
              <br />
              <label
                htmlFor="review"
              >
                Review: 
                <br />
                <textarea
                  name="review"
                  placeholder="add review here"
                />
              </label>
              <br />
              <label
                htmlFor="link"
              >
                Link to Buy:
                <input
                  name="link"
                  type="url"
                />
              </label>
              <br />
              <select
                name="game_type"
              >
                <option
                  value=""
                >
                  Game Type:
                </option>
                <option
                  value="video"
                >
                  Video
                </option>
                <option
                  value="tabletop"
                >
                  Tabletop
                </option>
              </select>
            </fieldset>
            <button
              type="submit"
            >
              Add Review
            </button>
          </form>
        </section>
        <h2>
          Review List
        </h2>
        <withRouter(AdminReview)
          key="288"
          review={
            Object {
              "game_type": "tabletop",
              "link": "http://www.notArealSite.com",
              "picture": "['http://ecx.images-amazon.com/images/I/51l4YWeAEvL._SY300_.jpg']",
              "review": "This was the best game ever!",
              "title": "Test Review 1",
            }
          }
        />
        <withRouter(AdminReview)
          key="288"
          review={
            Object {
              "game_type": "tabletop",
              "link": "http://www.notArealSite.com",
              "picture": "['http://ecx.images-amazon.com/images/I/51l4YWeAEvL._SY300_.jpg']",
              "review": "This was an ok game.",
              "title": "Test Review 2",
            }
          }
        />
        <Pagination
          items={
            Array [
              Object {
                "game_type": "tabletop",
                "link": "http://www.notArealSite.com",
                "picture": "['http://ecx.images-amazon.com/images/I/51l4YWeAEvL._SY300_.jpg']",
                "review": "This was the best game ever!",
                "title": "Test Review 1",
              },
              Object {
                "game_type": "tabletop",
                "link": "http://www.notArealSite.com",
                "picture": "['http://ecx.images-amazon.com/images/I/51l4YWeAEvL._SY300_.jpg']",
                "review": "This was an ok game.",
                "title": "Test Review 2",
              },
            ]
          }
          page={0}
          pageLimit={10}
          setPage={[Function]}
        />
      </withRouter(AdminReviewList)>
    </ContextProvider>
  </withRouter(App)>
</Router>
`;
